### DISCUSSION TITLE
Choose organization on login

### DISCUSSION THREAD

**Discussion by eliapimpinicchio** [zitadel/zitadel#8700] [Q&A] [answered]
Hello everyone,

I have the following situation: 
A user is part of several organizations. 
Upon login, the user should choose which organization to access, as each organization has different software.

How can I retrieve the list of all organizations for a given user?

**Comment by hifabienne** [comment]
The following request returns the list of organizations, for the requesting app and user: https://zitadel.com/docs/apis/resources/auth/auth-service-list-my-project-orgs

**Reply by eliapimpinicchio** [reply]
Thank you very much for your response.

However, from the documentation I read that the request works once a project has already been selected.

My workflow should be: 
1- User logs in 
2- I retrieve all the organizations in which the logged-in user is present 
3- User selects an organization 
4- I retrieve all the projects of that organization 
5- User selects the project

So my question is whether it is possible to retrieve all the organizations of a user without selecting a project.

**Reply by hifabienne** [reply, marked as answer]
In that case i think you kind of need to build it yourself.
1. Get authorizations/grants of a user: https://zitadel.com/docs/apis/resources/auth/auth-service-list-my-user-grants
2. Read organizations by org id, to be able to present names
3. Show only those projects for the specific organization (based on the frist request aswell)

**Reply by eliapimpinicchio** [reply]
Hi, with your flow, I can now retrieve the organizations, projects, and related applications.

Now, for example, if I want to log into an application, how can I perform this action without requiring a second login to retrieve the ID token (for example silent login inside target application)?

My application flow is as follows:
1- The user logs in with email and password.
2- The user selects an organization.
3- The user selects an application, strarting from project search.
4- The selected application opens in a new browser tab.

Can you suggest a solution?

**Reply by hifabienne** [reply]
you can send the prompt: none in the authorization request, in that case the user already needs to be authenticated without any interaction, you should also send the id_token_hint in that case.
you can find the information here: https://zitadel.com/docs/apis/openidoauth/endpoints#additional-parameters

**Reply by eliapimpinicchio** [reply]
Ok, but in which way can i take the id_token?
For take organizations, projects and applications i use the session token.

In particular the requests flow is the following:
1-  https://:DOMAIN/v2/sessions + service admin token: with login name for verify if a user exists
2- https://:DOMAIN/v2/sessions/${req.body.sessionId} + service admin token: with password for login and have the session token
3- https://:DOMAIN/auth/v1/usergrants/me/_search(session token): for retrive projects
4- https://:DOMAIN/management/v1/projects/:projectId/apps/_search (service token ): for retrive applications inside projects

**Reply by hifabienne** [reply]
Ah, you are using the session API, do you build you own login ui? 
The session tokens are not oidc compliant, they are opaque tokens, and you have to interpret them yourself.
If you are building your own login ui you also have to build the oidc standard if you like to get the id and access tokens.
You can find some information here: https://zitadel.com/docs/guides/integrate/login-ui/oidc-standard

Otherwise you can use our hosted login. wich is automatically deplyed.
You can find the wellknown endpoint here: your-domain/.well-known/openid-configuration

**Comment by ginsul** [comment]
Hi @eliapimpinicchio, @hifabienne 
Sorry for jumping in. Have you resolved your issue?
I had a similar problem where I needed to select an organization when logging in.
